lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,2106 @@
+                        alias))
+            if config is None:
+                config = self.__cloudfront_facts_mgr.get_streaming_distribution_config(
+                    streaming_distribution_id).get('StreamingDistributionConfig')
+            if e_tag is None:
+                e_tag = self.__cloudfront_facts_mgr.get_etag_from_distribution_id(
+                    streaming_distribution_id, True)
+            return streaming_distribution_id, config, e_tag
+        except Exception as e:
+            self.module.fail_json(
+                msg="error validating parameters for streaming distribution " +
+                "update and delete - " + str(e))
+
+    def validate_tagging_arn(self, alias, distribution_id,
+                             streaming_distribution_id):
+        try:
+            distribution_response = None
+            streaming_distribution_response = None
+            if(alias is not None and (distribution_id is not None or
+               streaming_distribution_id is not None)):
+                self.module.fail_json(
+                    msg="both alias and a distribution id have been specified" +
+                    "for tagging a resource. please only specify one.")
+            if(distribution_id is not None and
+               streaming_distribution_id is not None):
+                self.module.fail_json(
+                    msg="both distribution_id and streaming_distribution_id " +
+                    "have been specified. please only specify one.")
+            if alias is not None:
+                distribution_id = (
+                    self.__cloudfront_facts_mgr.get_distribution_id_from_domain_name(
+                        alias))
+            if distribution_id is not None:
+                distribution_response = (
+                    self.__cloudfront_facts_mgr.get_distribution(
+                        distribution_id))
+            if distribution_response is not None:
+                distribution = distribution_response.get('Distribution')
+                if distribution is not None:
+                    return distribution.get('ARN')
+            streaming_distribution_response = (
+                self.__cloudfront_facts_mgr.get_streaming_distribution(
+                    streaming_distribution_id))
+            if streaming_distribution_response is not None:
+                streaming_distribution = streaming_distribution_response.get(
+                    'StreamingDistribution')
+                if streaming_distribution is not None:
+                    return streaming_distribution.get('ARN')
+            self.module.fail_json(
+                msg="unable to find a matching distribution with " +
+                "given parameters")
+        except Exception as e:
+            self.module.fail_json(
+                msg="error validating tagging parameters - " + str(e) +
+                "\n" + traceback.format_exc())
+
+ 
```

Comment:
This module is not idempotent if you can't run it twice with the same arguments. 

If you have to know the distribution ID to update it, but not create it, then this is not going to work as it should. Really it should be searching for a distribution that matches the unique distribution properties (presumably the origins of the distribution)